app/objects/c_agent.py

```diff
@@ -137,19 +137,40 @@ async def calculate_sleep(self):
         return self.jitter('%d/%d' % (self.sleep_min, self.sleep_max))
 
     async def capabilities(self, ability_set):
+        """Get abilities that the agent is capable of running
+
+        :param ability_set: List of abilities to check agent capability
```

Comment:
nit: the variable name implies a `set` but it's a `list`.  consider renaming to be type agnostic (it looks like this method only cares that it's iterable, but i'm not sure if it's assumed that the members of `ability_set` are unique).